src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -826,5 +826,90 @@ namespace OpenLoco::Vehicles
                 regs.esi = X86Pointer(newComponent);
                 return 0;
             });
+
+        registerHook(
+            0x004AF5E1,
+            [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
+                registers backup = regs;
+                VehicleHead* head = X86Pointer<VehicleHead>(regs.esi);
+                connectJacobsBogies(*head);
+                regs = backup;
+                return 0;
+            });
+    }
+
+    // remember to move this above registerHooks after rebasing
+    // 0x004AF5E1
+    // esi: head
+    // returns nothing
+    void connectJacobsBogies(VehicleHead& head)
+    {
+        // if jacob's bogie is available at front the front body will have jacobsBogieAvailable
+        // if jacob's bogie is available at back the end body will have jacobsBogieAvailable
+        Vehicle train(head);
+        auto componentsFound = 0;
+        CarComponent previousCarComponent;
+        CarComponent secondPreviousCarComponent;
+        for (auto& car : train.cars)
+        {
+            for (auto& component : car)
+            {
+                if (component.body->has38Flags(Flags38::jacobsBogieAvailable))
+                {
+                    // reset jacob's bogie at the front of the car
+                    if (component.body->getSubType() == VehicleEntityType::body_start)
+                    {
+                        component.front->objectSpriteType = 0xFF;
+                        component.back->objectSpriteType = 0xFF;
+                        component.body->objectSpriteType = 0xFF;
+                        auto frontBogieOfNext = component.body->nextVehicleComponent();
+                        if (frontBogieOfNext == nullptr)
+                        {
+                            throw Exception::RuntimeError("connectJacobsBogies frontBogieOfNext was unexpectedly nullptr");
+                        }
+
+                        CarComponent nextComponent = CarComponent(frontBogieOfNext);
+                        auto o = ObjectManager::get<VehicleObject>(component.front->objectId);
+                        nextComponent.front->objectSpriteType = o->carComponents[nextComponent.body->bodyIndex].frontBogieSpriteInd;
+                        if (nextComponent.body->has38Flags(Flags38::isReversed))
+                        {
+                            nextComponent.front->objectSpriteType = o->carComponents[nextComponent.body->bodyIndex].backBogieSpriteInd;
+                        }
+                        // connect jacob's bogie to the front
+                        if (componentsFound >= 0 and previousCarComponent.body->has38Flags(Flags38::jacobsBogieAvailable))
```

Comment:
although legal to use `and` always looks out of place in a c++ codebase. Prefer `&&`